flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -3689,9 +3739,12 @@ class ArrayExprLowering {
     determineShapeOfDest(lhs);
     semant = ConstituentSemantics::RefTransparent;
     auto exv = lowerArrayExpression(rhs);
-    builder.create<fir::ArrayMergeStoreOp>(
-        loc, destination, fir::getBase(exv), destination.memref(),
-        destination.slice(), destination.typeparams());
+    if (explicitSpaceIsActive())
+      builder.create<fir::ResultOp>(loc, fir::getBase(exv));
+    else
+      builder.create<fir::ArrayMergeStoreOp>(
+          loc, destination, fir::getBase(exv), destination.memref(),
+          destination.slice(), destination.typeparams());
```

Comment:
Is the `array_merge_store` being created outside of the outermost explicit loop somewhere else?

Reply:
Yes, it is created by the explicit context handling just like in `lowerArrayAssignment` [here](https://github.com/flang-compiler/f18-llvm-project/blob/ae9218ca2354c0cb62d77b7a529632f486102469/flang/lib/Lower/ConvertExpr.cpp#L3106-L3111).